packages/textInput/components/TextInputWithButtons.tsx

```diff
@@ -25,7 +25,12 @@ export interface TextInputWithButtonsProps
   buttons: Array<React.ReactElement<TextInputButtonProps>>;
 }
 
-const TextInputWithButtons = (props: TextInputWithButtonsProps) => {
+const TextInputWithButtons = ({
+  type = "text",
+  appearance = InputAppearance.Standard,
+  showInputLabel = true,
+  ...props
+}: TextInputWithButtonsProps) => {
   const [hasFocus, setHasFocus] = React.useState(false);
```

Comment:
Sorry @derekhassan we've actually noticed another issue with this after some downstream testing. 
There is some broken functionality here. 

If you compare this story in this PR:
<img width="862" alt="Screenshot 2023-03-17 at 2 15 10 PM" src="https://user-images.githubusercontent.com/34781875/225998726-67989b26-6838-4ca2-88e0-af6181c74441.png">

You will notice it is missing that icon start in this component. It should look like this:
<img width="740" alt="Screenshot 2023-03-17 at 2 16 34 PM" src="https://user-images.githubusercontent.com/34781875/225999043-95146e71-14a1-4ad2-bbf8-b041c1cac33a.png">

Reply:
Ah, apologies for the broken icon, it looks like I had removed the extra non-destructured props from `getIconStartContent` 😅. I looked into the issue about being unable to change the appearance and it seems to be an issue with the `TextInputWithButtons` story. I think this behavior is related to [this Storybook issue](https://github.com/storybookjs/storybook/issues/17025). If I move the buttons array outside the function, the issue goes away. I'm not sure if this is the ideal solution and was wondering if you had any feedback on this approach? (This is in [TextInputWithButtons.stories.tsx](https://github.com/d2iq/ui-kit/blob/main/packages/textInput/stories/TextInputWithButtons.stories.tsx))

```jsx
const buttonsPlaceholder = [
  <TextInputButton
    key={0}
    color="red"
    shape={SystemIcons.Close}
    onClick={btnClickFn}
    aria-label="Clear input"
  />
];

export const WithCustomIcons = args => {
  const { buttons, ...rest } = args;
  return (
    <TextInputWithButtons
      id="withIcon.colored"
      inputLabel="With colored icon"
      iconStart={<Icon shape={SystemIcons.Search} />}
      buttons={
        buttons && Object.values(buttons).length ? buttons : buttonsPlaceholder
      }
      {...rest}
    />
  );
};
```

Reply:
It looks like some of these issues may be resolved with Storybook 7 which we plan to upgrade to soon. 
Go ahead and fix the other issues with the component and its props and if there is still an issue with the appearance, don't worry about it. I'll do some research and see how I can resolve that. It's more important that the component functions as expected and I have seen some issues myself with the controls for the stories and plan to improve those in the near future. Thanks for your research on this matter!

Reply:
Gotcha, thanks!